packages/language/src/syntax-tree/ast-iterator.ts

```diff
@@ -931,3 +931,53 @@ export function forEachNode(
       break;
   }
 }
+
+export function traverseAllNodes(node: SyntaxNode, traverse: (n: SyntaxNode) => void) {
+  traverse(node);
+  forEachNode(node, (child) => {
+    traverseAllNodes(child as SyntaxNode, traverse);
+  });
+}
+
+/**
+ * Checks whether the given syntax tree contains at least one node of the specified kind.
+ *
+ * @param root - The root node to start searching from.
+ * @param kind - The syntax kind to search for.
+ * @returns True if a matching node is found, false otherwise.
+ */
+export function containsNodeOfKind(
+  root: SyntaxNode,
+  kind: SyntaxKind,
+): boolean {
```

Comment:
Suggestion: Can we make this into a `find` instead? That is much more useful than the binary info whether something exists, even though it requires the same amount of work.
```suggestion
export function findFirstNode<T>(
  root: SyntaxNode,
  kind: SyntaxKind,
): T | undefined {
```

Reply:
Sure thing!